### PR TITLE
Adding unit tests to SetStatuses() method in Service class

### LINKED ISSUE #9015: Add unit tests for Service and Repository class needed for Bulk status update
Part of #1923 

#### Describe the solution presented in this issue.
Add unit tests for Service and Repository class needed for Bulk status update
 
Also, Today, the application validation of a single status update is happening in the controller class. Move the validation to the service class. Controller should not carryout the validation.

### PR DESCRIPTION
**Pull request by swatkat1**
### Description

Adding unit tests to SetStatuses() method in Service class.

### Issue(s) this completes

Fixes #9015

### CODE REVIEW

File: server/test/services/applications/ProgramAdminApplicationServiceTest.java [hunk 1]
```diff
@@ -361,6 +570,76 @@ public void setStatus_tiApplicant_sendsEmail() throws Exception {
                 STATUS_WITH_ONLY_ENGLISH_EMAIL.localizedEmailBodyText().get().getDefault()));
   }
 
+  @Test
+  public void setStatuses_tiApplicant_sendsEmail_nonDefaultLocale() throws Exception {
+    String userEmail = "user@email.com";
+    String tiEmail = "ti-ko@email.com";
+    EmailSendClient emailSendClient = Mockito.mock(EmailSendClient.class);
+    MessagesApi messagesApi = instanceOf(MessagesApi.class);
```
Comment (ssandbekkhaug) [inline server/test/services/applications/ProgramAdminApplicationServiceTest.java:578]: Suggestion (non-blocking): The code to create the service is repeated several times in different tests. Would it make sense to make a helper method?

**Review by ssandbekkhaug — APPROVED** [review]

File: server/test/services/applications/ProgramAdminApplicationServiceTest.java (hunk 1 above)
Comment (swatkat1) [reply server/test/services/applications/ProgramAdminApplicationServiceTest.java:523]: sure!

Comment (swatkat1) [reply server/test/services/applications/ProgramAdminApplicationServiceTest.java:523]: unfortunately, emailSendClient is needed for the verify() step of the steps.

Comment (swatkat1) [reply server/test/services/applications/ProgramAdminApplicationServiceTest.java:523]: I did push the code creating the service with mocked object to helper method

File: server/app/services/applications/ProgramAdminApplicationService.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
+import autovalue.shaded.com.google.common.annotations.VisibleForTesting;
```
Comment (gwendolyngoetz) [inline server/app/services/applications/ProgramAdminApplicationService.java:5]:
```suggestion
import com.google.common.annotations.VisibleForTesting;
```
(blocking) Oh looks like the wrong package name was used.

File: server/test/services/applications/ProgramAdminApplicationServiceTest.java [hunk 3]
```diff
@@ -107,12 +110,49 @@ public void getApplication() {
...
+    ImmutableList<Long> appIdList = createApplicationList(3, program);
+
+    var result = service.getApplications(appIdList, program.getProgramDefinition());
+    assertThat(result).isNotEmpty();
+    assertThat(result.size()).isEqualTo(3);
+
+    result.stream()
+        .forEach(
+            e -> {
+              assertThat(appIdList.contains(e));
```
Comment (gwendolyngoetz) [inline server/test/services/applications/ProgramAdminApplicationServiceTest.java:126]:
```suggestion
              assertThat(appIdList).contains(e);
```
(blocking) This isn't actually asserting anything. I think this is what you want.

**Review by gwendolyngoetz — DISMISSED** [review]

**Review by ssandbekkhaug — APPROVED** [review]

**Comment by swatkat1** [comment]
@gwendolyngoetz  - Can you please review to approve? Thanks!

**Pull request merged**